### DISCUSSION TITLE
[Question][Hw6] Resources regarding fuzzing

### DISCUSSION THREAD

**Discussion by m-spitfire** [prosyslab-classroom/cs348-information-security#719] [Q&A]
Name: Murad Bashirov

Hello. Is there any recommended paper/book chapter regarding mutation based fuzzing for string inputs? Talking about different mutation operators and evaluating them. I tried to search for it and not many of them talk about what kind of operators there are. Generally there're

- Adding character
- Removing character
- Flipping character

But is that it? Thanks.

**Comment by KihongHeo** [comment]
Hi. 
There is no rule of thumb. You can define whatever you want.

Maybe the fuzzing book I put in the course webpage is a good starting point.
